src/mvs_eland/C0_data_processing.py

```diff
@@ -1696,35 +1699,49 @@ def add_maximum_cap(dict_values, group, asset, subasset=None):
     asset: str
         asset name
     subasset: str
-        subasset name
+        subasset name.
+        Default: None.
 
     Returns
     -------
 
     """
     if subasset is None:
-        dict = dict_values[group][asset]
+        asset_dict = dict_values[group][asset]
     else:
-        dict = dict_values[group][asset][subasset]
-    if MAXIMUM_CAP in dict:
-        # check if maximumCap is greater that installedCap
-        if dict[MAXIMUM_CAP][VALUE] is not None:
-            if dict[MAXIMUM_CAP][VALUE] < dict[INSTALLED_CAP][VALUE]:
+        asset_dict = dict_values[group][asset][subasset]
+    if MAXIMUM_CAP in asset_dict:
+        # Check if a maximumCap is defined
+        if asset_dict[MAXIMUM_CAP][VALUE] is not None:
```

Comment:
When would "asset_dict[MAXIMUM_CAP][VALUE] == None" be True? When no value is inserted in the csv? Should there be a warning message for this?

Reply:
I only added 
```
# adapt maximumCap of non-dispatchable sources
            if (
                group == ENERGY_PRODUCTION
                and asset_dict[FILENAME] != None
                and max(asset_dict[TIMESERIES_NORMALIZED]) != 1
            ):
                asset_dict[MAXIMUM_CAP][VALUE] = asset_dict[MAXIMUM_CAP][VALUE] / max(
                    asset_dict[TIMESERIES_NORMALIZED]
                )
                logging.debug(
                    f"Parameter {MAXIMUM_CAP} of asset '{asset_dict[LABEL]}' was divided by the peak value of {TIMESERIES_NORMALIZED} as it did not equal 1. This was done as the aimed constraint is to limit the power, not the flow."
                )
```
I did not check the code that was there before. I think it makes sense to open a new issue for such things as this PR would get to long.
Again, why this part is green is because you asked me to change `dict` into another variable name.